sklearn/ensemble/iforest.py

```diff
@@ -206,7 +219,9 @@ def predict(self, X):
 
         depths += _average_path_length(n_samples_leaf)
 
-        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
Yes seems like we forgot to remove it from the doc, we didn't need it anymore. But we may need it now, since as you said base_estimator don't know `max_samples`...